tests/unit/test_adapter.py

```diff
@@ -198,7 +205,7 @@ def test_measurements_adapter_ec_model(eciML1515):
         growth_rate,
     )
     # 4 operations (1 protein + 1 uptake + 1 secretion + growth rate) + 3 warnings
-    # (1 skipped protein + 2 removed proteins) are expected:
+    # (1 skipped protein + 2 removed proteins) + 1 error (unmatched rate) are expected:
```

Comment:
These numbers have grown a bit, I think now it would make sense to test each case in isolation. It's more verbose and the coverage would be the same, but if one of the cases fails, you could more quickly pinpoint the cause. Currently, one of the numbers wouldn't match and you'd have to do some debugging to figure out which expectation didn't match the result.

Reply:
With [`1a2eff4`](https://github.com/DD-DeCaF/simulations/pull/151/commits/1a2eff4ad40df19b3aa2b39b0fa1dfa2eb5f020c) and [`45433fe`](https://github.com/DD-DeCaF/simulations/pull/151/commits/45433fe9029edf9474ee0ef5f6df543448daef68) I have now split this test in 5 different ones to isolate all cases:

* 3 in `adapter`:
  * `test_measurements_adapter_errors`: tests on a normal model that unmatched metabolites (for uptake/secretion rate) or unmatched reactions (for fluxomics) are recorded as errors. The latter of the two was not done before so we increased a bit in coverage :)
  * `test_measurements_adapter_ec_model`: the original one, now only controlling that proteomics + uptake/secretion rates + growth rate are adequately added to an ecModel.
  * `test_measurements_adapter_ec_model_skip`: tests that the flex. algorithm is skipped when no growth rate is provided + tests that proteins not in ecModel are saved as warnings.
* 2 in `driven`:
  * `test_flexibilize_proteins`: tests that the flex. algorithm removes at least 2 proteins properly if the values are too low.
  * `test_flexibilize_proteins_skip`: tests that the flex. algorithm is skipped if an uptake/secretion rate has a non-existing metabolite (see [`1b1b11d`](https://github.com/DD-DeCaF/simulations/pull/151/commits/1b1b11dcc812c0dd0538865a5036496e421e57d9)).

As you mentioned, it is a little more verbose and there's some repetition, but hopefully it will be easier to debug in the future.

Reply:
Thank you, the tests look very clear and on the point now!